app/models/mark.rb

```diff
@@ -37,6 +37,11 @@ def valid_mark
     end
   end
 
+  def scale_mark(curr_max_mark, prev_max_mark)
+    new_mark = self.mark * (curr_max_mark.to_f / prev_max_mark)
```

Comment:
This actually needs to be implemented separately for each criterion type (this helps avoid potential floating point errors).
Note that for a Rubric criterion, the value of `(mark / max_mark) * 4` should always be an *integer* between 0 and 4, and for a Checkbox criterion, the value of `mark` should always be either 0 or `max_mark`.

The logic you've written here is only appropriate for Flexible Criterion (where an arbitrary scaling is necessary). However, in this case you should round the mark to two decimal places.

Finally, for all three criterion types, you should do a check for `prev_max_mark` being 0; in this case, no scaling should occur.